src/components/Auth/Register.js

```diff
@@ -0,0 +1,394 @@
+
+        break;
+    };
+  }
+
+  inputChecking(event) {
+    const name = event.target.name;
+
+    switch(name) {
+      case "email":
+        if(/@ttu.edu$/.test(this.state.email) !== true) {
+          this.setState({
+            err: update(this.state.err, {
+              emailErr: {$set: 'Please use a TTU email address.'}
+            })
+          });
+        }
+
+        break;
+      case "password":
+        if(/^[\x00-\x7F]+$/.test(this.state.password) !== true) {
+          // ASCII only
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use only ASCII characters.'}
+            })
+          });
+        } else if(this.state.password.length < 8) {
+          // 8 characters long
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use at least 8 characters.'}
+            })
+          });
+        } else if(/[A-Z]/.test(this.state.password) !== true) {
+          // 1 uppercase
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use at least one uppercase letter.'}
+            })
+          });
+        } else if(/[a-z]/.test(this.state.password) !== true) {
+          // 1 lowercase
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use at least one lowercase letter.'}
+            })
+          });
+        } else if(/[0-9]/.test(this.state.password) !== true) {
+          // 1 number
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use at least one number.'}
+            })
+          });
+        } else if(/[^A-Za-z0-9]/.test(this.state.password) !== true) {
+          // 1 special character
+          this.setState({
+            err: update(this.state.err, {
+              passErr: {$set: 'Please use at least one special character.'}
+            })
+          });
+        }
+
+        break;
+      case "confirmPass":
+        if(this.state.password !== this.state.confirmPass) {
+          this.setState({
+            err: update(this.state.err, {
+              confirmErr: {$set: 'Please enter a matching password.'}
+            })
+          });
+        }
+
+        break;
+      case "studentID":
+        if(/^[0-9]{8}$/.test(this.state.studentID) !== true) {
+          this.setState({
+            err: update(this.state.err, {
+              stuIDErr: {$set: 'Please use 8 numbers.'}
+            })
+          });
+        }
+
+        break;
+      default:
+        console.log("Create an onBlur handler for " + name);
+
+        break;
+    };
+  }
+
+  render() {
+    var studentIDdesc = null;
+    var studentIDField = null;
+    if(this.state.role === 'student') {
+      studentIDdesc = (
+        <Paragraph>
+          Don't include the R before your student ID number.
```

Comment:
It's probably complaining because the file ends with `.js` rather than `.jsx` even though it uses JSX syntax